Update diagram in subplots_adjust documentation to clarify parameters

Bindi003 opened the pull request:
## PR summary
This pull request addresses an issue in the documentation for [](https://matplotlib.org/stable/api/_as_gen/matplotlib.figure.Figure.subplots_adjust.html#matplotlib.figure.Figure.subplots_adjust) where the diagram incorrectly implied that the left, right, top, and bottom parameters were distances from the nearest subplot edges. In reality, these parameters represent distances from the edges of the figure (the overall plotting area).

- Why is this change necessary?
The current diagram is misleading and contradicts the correct usage of subplots_adjust, potentially causing confusion for users trying to control subplot layout. 

- What is the reasoning for this implementation?
The implementation involves updating the Python script (doc/_embedded_plots/figure_subplots_adjust.py) that generates the diagram embedded in the documentation. The plotting code has been modified to accurately represent the relationship between the subplots_adjust parameters and the figure edges, matching the function's actual behavior.

## PR checklist

- [ ]  closes #30022 [matplotlib.figure.Figure.subplots_adjust figure](https://github.com/matplotlib/matplotlib/issues/30022#issuecomment-2861297324)
- [N/A ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in [figure_subplots_adjust.py](https://github.com/matplotlib/matplotlib/blob/main/doc/_embedded_plots/figure_subplots_adjust.py)
- [N/A ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

QuLogic commented:
[This appears incorrect](https://output.circle-artifacts.com/output/job/24d752b0-090a-47ea-a51c-e91f896f8dcd/artifacts/0/doc/build/html/api/_as_gen/matplotlib.figure.Figure.subplots_adjust.html):
![image](https://github.com/user-attachments/assets/f1aac410-465c-4c01-9cbc-f61a5b476f13)
The origin for all the parameters it the bottom left, not the top right.

timhoffm commented:
Yes should be something like this ![image](https://github.com/user-attachments/assets/ef776f00-5169-4332-a1de-7e71a7aa28bd)

Bindi003 commented:
Hi @QuLogic @timhoffm I've addressed the feedback on the diagram and pushed the updated code. Hopefully, it should now be correct. This is my first contribution, so I appreciate your guidance! Thank you!

Bindi003 commented on doc/_embedded_plots/figure_subplots_adjust.py:
```diff
@@ -17,14 +18,14 @@ def arrow(p1, p2, **props):
 arrow((0, 0.75), (0.1, 0.75))  # left
 arrow((0.435, 0.25), (0.565, 0.25))  # wspace
-arrow((0.1, 0.8), (1, 0.8))  # right
+arrow((0, 0.8), (0.9, 0.8))  # right
```
Hi @story645 , thanks for the suggestion! I've updated the code to use the direct overlay.annotate() calls as you recommended :)

timhoffm replied:
@story645 I disagree. I intentionally introduced the arrow function as a semantic wrapper. I still find `annotate("", xy1, xy2)` a really awkward API. This is basically the discussion of #29826 (which is only deferred because we need to agree what the underlying artist should be), and the need for a reasonable high-level API still persists.

I'm not quite happy with reverting the wrapper, but since this is only an internal helper script and does not show up prominently in the docs, it's not worth fighting over.

timhoffm replied:
This is not intoducing new API and the code is not user-facing or teaching by intent. In my perspective, it's just a [helper function](https://matplotlib.org/devdocs/users/explain/quick_start.html#making-a-helper-functions) making the example *more* readable for editors of that code. Obviously, YMMV. Let's not get into the discussion right now.

timhoffm commented:
Thanks @Bindi003 and congratulations on your first contribution to Matplotlib 🎉! We'd be happy to see you again.
